test/functional/wallet_pruning.py

```diff
@@ -0,0 +1,132 @@
+            [], # node dedicated to mining
+            ['-prune=550'], # node dedicated to testing pruning
+        ]
+
+    def setup_nodes(self):
+        self.add_nodes(self.num_nodes, extra_args=self.extra_args)
+        self.start_node(0)
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def create_big_chain(self):
+        self.log.info("Generating a long chain of blocks...")
+        # Generate 288 light blocks, the minimum required to stay on disk with prune enabled
+        # do not sync
+        self.generate(self.nodes[0], 288, sync_fun=lambda: None)
+
+        # Generate large blocks to make sure we have enough to test chain pruning
+        self.mine_large_blocks(self.nodes[0], 600)
+
+    def test_wallet_import_pruned_test(self):
+        self.log.info("Make sure we can import wallet when pruned and required blocks are still available")
+
+        wname = "wallet_pruned.dat"
+
+        # export wallet
+        self.nodes[1].dumpwallet(os.path.join(self.nodes[1].datadir, wname))
+        # import wallet
+        self.nodes[1].importwallet(os.path.join(self.nodes[1].datadir, wname))
+
+        # mine some blocks, pruning should not have removed the block
+        self.mine_large_blocks(self.nodes[0], 5)
+        self.sync_all()
+
+        # import wallet, should still work
+        self.nodes[1].importwallet(os.path.join(self.nodes[1].datadir, wname))
+        self.log.info("Wallet successfully imported on pruned node")
+
+    def test_wallet_import_pruned_with_missing_blocks(self):
+        self.log.info("Make sure we cannot import wallet when pruned and required blocks are not available")
+
+        wname = "wallet_pruned_missing_blocks.dat"
+        # export wallet
+        self.nodes[1].dumpwallet(os.path.join(self.nodes[1].datadir, wname))
+
+        # get birthheight of wallet
+        with open(os.path.join(self.nodes[1].datadir, wname), 'r', encoding="utf8") as f:
+            for line in f:
+                if line.startswith('# * Best block at time of backup'):
+                    wallet_birthheight = int(line.split(' ')[9])
+                    break
+
+        # mine large blocks, blocks will be pruned automatically
+        self.mine_large_blocks(self.nodes[0], 600)
+        self.sync_all()
+
+        assert_raises_rpc_error(-1, "Block not available (pruned data)", self.nodes[1].getblock, self.nodes[1].getblockhash(wallet_birthheight))
+
+        # make sure wallet cannot be imported because of missing blocks
+        assert_raises_rpc_error(-4, "Pruned blocks from height 876 required to import keys. Use RPC call getblockchaininfo to determine your pruned height.", self.nodes[1].importwallet, os.path.join(self.nodes[1].datadir, wname))
+
+    def run_test(self):
+        self.log.info("Warning! This test requires ~1.5GB of disk space")
+
+        self.create_big_chain()
+
+        # connect mid-test node1
```

Comment:
nit: maybe add some comments here to explain the rationale behind this mid-test connection? I think it will not be very clear for someone reading this test in the future.